pyproject.toml

```diff
@@ -190,7 +190,18 @@ exclude = [
     "localstack-core/.filesystem",
     ".git",
     "localstack-core/localstack/services/stepfunctions/asl/antlr/runtime"
-    ]
+]
+
+[tool.ruff.per-file-target-version]
+# Only allow minimum version for code used in the CLI
+"localstack-core/localstack/cli/**" = "py39"
+"localstack-core/localstack/packages/**" = "py39"
+"localstack-core/localstack/config.py" = "py39"
+"localstack-core/localstack/constants.py" = "py39"
+"localstack-core/localstack/utils/analytics/**" = "py39"
+"localstack-core/localstack/utils/bootstrap.py" = "py39"
+"localstack-core/localstack/utils/json.py" = "py39"
```

Comment:
I'm unsure whether this set of files is correct or maybe even too much (e.g., in the case of the analytics code where I even had to remove a `match` statement)